build/pxc-entrypoint.sh

```diff
@@ -488,6 +491,55 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 		fi
 		rm "$wsrep_verbose_logfile"
 	fi
+	if [ -n "$PXC_SERVICE" ]; then
+		function get_primary() {
+			peer-list -on-start=/usr/bin/get-pxc-state -service="$PXC_SERVICE" 2>&1 \
+				| grep wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary \
+				| sort \
+				| tail -1 \
+				|| true
+		}
+		function node_recovery() {
+			set -o xtrace
+			echo "Recovery is in progress, please wait...."
+			sed -i 's/wsrep_cluster_address=.*/wsrep_cluster_address=gcomm:\/\//g' /etc/mysql/node.cnf
+			rm -f /tmp/recovery-case
+			if [ -s "$grastate_loc" ]; then
+				sed -i 's/safe_to_bootstrap: 0/safe_to_bootstrap: 1/g' "$grastate_loc"
+			fi
+			echo "Recovery was finished."
+			exec "$@" $wsrep_start_position_opt
+		}
+
+		is_primary_exists=$(get_primary)
+		if [[ -z "$is_primary_exists" && -f "$grastate_loc" && $safe_to_bootstrap != 1 ]] || [[ -z "$is_primary_exists" && -f "${DATADIR}/gvwstate.dat" ]]; then
```

Comment:
```suggestion
		if [[ -z "$is_primary_exists" && $safe_to_bootstrap != 1 ]] || [[ -z "$is_primary_exists" && -f "${DATADIR}/gvwstate.dat" ]]; then
```
why it is not enough?

Reply:
@delgod  In case of recovery from backup we do not have 'primary'  and we do not have '$safe_to_bootstrap' and we get the recovery loop. So we need to check whether 'grastate.dat' exists (in full crash it should exist) or we need to check whether $safe_to_bootstra is empty or not.

Reply:
coz sometimes `safe_to_bootstrap` is empty, which causes incorrect full crash entering